src/cursor/cur_layered.c

```diff
@@ -265,27 +269,42 @@ __clayered_open_cursors(WT_CURSOR_LAYERED *clayered, bool update)
     }
 
     if (clayered->stable_cursor == NULL) {
-        leader = S2C(session)->layered_table_manager.leader;
-        if (!leader)
+        leader = conn->layered_table_manager.leader;
+        stable_uri = layered->stable_uri;
+        if (!leader) {
             /*
              * We may have a stable chunk with no checkpoint yet. If that's the case then open a
              * cursor on stable without a checkpoint. It will never return an invalid result (it's
              * content is by definition trailing the ingest cursor. It is just slightly less
              * efficient, and also not an accurate reflection of what we want in terms of sharing
              * checkpoint across different WiredTiger instances eventually.
              */
-            ckpt_cfg[cfg_pos++] = ",raw,checkpoint_use_history=false,force=true";
+
+            /* Look up the most recent data store checkpoint. This fetches the exact name to use. */
+            checkpoint_name = NULL;
+            WT_RET_NOTFOUND_OK(
+              __wt_meta_checkpoint_last_name(session, stable_uri, &checkpoint_name, NULL, NULL));
+
+            if (checkpoint_name == NULL) {
+                /* We've never picked up a checkpoint. */
```

Comment:
In this case, we open a normal dhandle?

Reply:
@quchenhao Yes, we get a regular btree on the name foo.wt_stable.  I _think_ this can only happen we start as a follower before a checkpoint ever happened.  There's a leader building oplog entries, and the follower is getting them and applying them to the ingest table.  But there's no content in the shared/stable btree (otherwise we'd have a checkpoint).  So we're basically opening an empty btree for the stable table.  It's a little wasteful because we don't need anything in that btree, but it's a corner case (that will be gone when the first checkpoint happens), and it makes the rest of the code easier.

Reply:
Since this PR is already merged, I'm adding this as a comment to my next PR using this file.